kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -233,155 +216,108 @@ define([
+         */
+        makeIframeFromSrc: function (arg) {
+            if (!arg.src && !arg.content) {
+                throw new Error('Report iframe must have either a source URL or direct content');
+            }
+            const iframe = html.tag('iframe'),
+                iframeId = 'frame_' + html.genId(),
+                height = arg.height || 'auto',
+                src = arg.src || 'data:text/html;charset=utf-8,' + encodeURIComponent(arg.content)
+            const iframeHtml = iframe({
+                class: this.baseCssClass + '__report_iframe',
+                style: {
+                    height: height
+                },
+                dataFrame: iframeId,
+                frameborder: '0',
+                scrolling: 'yes',
                 id: iframeId,
-                content: iframeHtml,
-            };
-        },
-
-        makeIframeSrc: function (arg) {
-            const t = html.tag,
-                iframe = t('iframe');
-
-            const iframeId = 'frame_' + html.genId();
-
-            const width = arg.width || '100%',
-                maxHeight = arg.maxHeight || 'auto',
-                iframeHtml = iframe({
-                    style: {
-                        display: 'block',
-                        width: width,
-                        height: arg.height,
-                        maxHeight: maxHeight,
-                        margin: 0,
-                        padding: 0,
-                    },
-                    dataFrame: iframeId,
-                    frameborder: '0',
-                    scrolling: 'yes',
-                    id: iframeId,
-                });
-
+                src: src
+            });
             return {
                 id: iframeId,
-                content: iframeHtml,
+                content: iframeHtml
             };
         },
 
-        escapeHtml: function (string) {
-            if (typeof string !== 'string') {
-                return;
-            }
-            const entityMap = {
-                '&': '&amp;',
-                '<': '&lt;',
-                '>': '&gt;',
-                '"': '&quot;',
-                "'": '&#39;',
-                '/': '&#x2F;',
-                '`': '&#x60;',
-                '=': '&#x3D;',
-            };
-            return String(string).replace(/[&<>"'`=/]/g, (s) => {
-                return entityMap[s];
-            });
-        },
-
         /**
          * Returns a Promise that resolves into a list of links to the hosted report
-         * @param {string} report
+         * @param {object} report
          */
         getLinks: function (report) {
             // NOTE: this returns a promise -- we must look up the html file set service url first.
-            const _this = this;
-
-            const client = new GenericClient({
-                url: Config.url('service_wizard'),
-                token: this.authToken(),
+            const serviceWizard = new DynamicServiceClient({
```

Comment:
Change to the other `DynamicServiceClient`, as its slightly more stable and testable.